### ISSUE TITLE
Casting Linux capabilities uint64 values to int will overflow on 32-bit builds

### ISSUE THREAD

**Issue by disconnect3d** [google/gvisor#5249]
**Description**

TL;DR: There are certain paths in the gVisor codebase, where fixed-size integers coming from linux capability values are casted to `int` and [`int`, similar to `uint` is either 32-bit or 64-bit in Go](https://golang.org/ref/spec#Numeric_types). The size of `int` depends on the architecture and so the `GOARCH` env variable when building Go projects.

Those cases looks like this: `bits.MaskOf64(int(...))` and can be seen here:

* https://github.com/google/gvisor/blob/a20da708291e2e5bdece5176dce61c1b4b10b7d9/pkg/sentry/kernel/auth/capability_set.go#L32
* https://github.com/google/gvisor/blob/a20da708291e2e5bdece5176dce61c1b4b10b7d9/pkg/sentry/kernel/auth/capability_set.go#L39
* https://github.com/google/gvisor/blob/a20da708291e2e5bdece5176dce61c1b4b10b7d9/runsc/specutils/specutils.go#L273
* https://github.com/google/gvisor/blob/a20da708291e2e5bdece5176dce61c1b4b10b7d9/runsc/container/container_test.go#L2222
* https://github.com/google/gvisor/blob/a20da708291e2e5bdece5176dce61c1b4b10b7d9/test/e2e/exec_test.go#L114

All of those cases operate on Linux capability values. On current systems, the maximum capabilities value, which refers to all capabilities, is `0000003fffffffff`. This value is bigger than 32-bits and so a 32-bit build of gVisor will overflow the value in the `int(cp)` or `int(cap)` expressions (from the code paths shown above) for certain Linux capability values.

The "affected" Linux capabilities are:
```
$ capsh --decode=0x0000003f00000000
0x0000003f00000000=cap_mac_override,cap_mac_admin,cap_syslog,cap_wake_alarm,cap_block_suspend,cap_audit_read
```

When an overflow with such value occurs, the resulting integer value is just 0. This can be seen here with the use of [`cint` Python module](https://github.com/disconnect3d/cint) in an IPython interactive shell:
```py
In [1]: from cint import I32

In [2]: I32(0x0000003f00000000)
Out[2]: I32(0)
```

It seems that this bug would make it impossible to enable capabilities which bits are bigger than 32 bits. Given this and the fact that gVisor [doesn't seem to support 32-bits builds yet(?)](https://github.com/google/gvisor/issues/5241) I decided to drop this here instead of reporting it as as security vuln.

Btw the issue also lies in the `bits.MaskOf64` template function (instantiated? in the [BUILD file](https://github.com/google/gvisor/blob/cb41f6703160dcb18242cd912db86983540af595/pkg/bits/BUILD#L28-L37), generated with the [`go_generics` tool](https://github.com/google/gvisor/tree/cb41f6703160dcb18242cd912db86983540af595/tools/go_generics)) defined in `pkg/bits/bits_template.go`:

https://github.com/google/gvisor/blob/a20da708291e2e5bdece5176dce61c1b4b10b7d9/pkg/bits/bits_template.go#L41-L44

This function takes `int` as a parameter, which will end up being  32-bit on 32-bits and so would not work as intended.

A fix should probably make this function take either `T` or just `uint64` as an argument? The same goes for the `Mask` function there.

**Steps to reproduce**

I haven't really tested this in practice as I didn't need to run gVisor in 32-bits. I just looked at some code here and there and noticed this.
